internal/tss/utils/transaction_service.go

```diff
@@ -0,0 +1,266 @@
+	}
+	// Wrap the transaction in a fee bump tx, signed by the distribution account
+	distributionAccountPublicKey, err := t.DistributionAccountSignatureClient.GetAccountPublicKey(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting distribution account public key: %w", err)
+	}
+
+	feeBumpTx, err := txnbuild.NewFeeBumpTransaction(
+		txnbuild.FeeBumpTransactionParams{
+			Inner:      tx,
+			FeeAccount: distributionAccountPublicKey,
+			BaseFee:    int64(t.BaseFee),
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("building fee-bump transaction %w", err)
+	}
+
+	feeBumpTx, err = t.DistributionAccountSignatureClient.SignStellarFeeBumpTransaction(ctx, feeBumpTx)
+	if err != nil {
+		return nil, fmt.Errorf("signing the fee bump transaction with distribution account: %w", err)
+	}
+	return feeBumpTx, nil
+}
+
+func (t *transactionService) parseErrorResultXDR(errorResultXdr string) (tss.RPCTXCode, error) {
+
+	//errorResult := xdr.TransactionResult{}
+	unMarshallErr := "unable to unmarshal errorResultXdr: %s"
+	//err := errorResult.UnmarshalBinary([]byte(errorResultXdr))
+
+	decodedBytes, err := base64.StdEncoding.DecodeString(errorResultXdr)
+	if err != nil {
+		return tss.RPCTXCode{OtherCodes: tss.UnMarshalBinaryCode}, fmt.Errorf(unMarshallErr, errorResultXdr)
+	}
+	dec := xdr3.NewDecoder(strings.NewReader(string(decodedBytes)))
+	var errorResult xdr.TransactionResult
+	_, err = dec.Decode(&errorResult)
+
+	if err != nil {
+		return tss.RPCTXCode{OtherCodes: tss.UnMarshalBinaryCode}, fmt.Errorf(unMarshallErr, errorResultXdr)
+	}
+	return tss.RPCTXCode{
+		TxResultCode: errorResult.Result.Code,
+	}, nil
+}
+
+func (t *transactionService) SendTransaction(transactionXdr string) (tss.RPCSendTxResponse, error) {
+	rpcResponse, err := t.sendRPCRequest("sendTransaction", map[string]string{"transaction": transactionXdr})
+	sendTxResponse := tss.RPCSendTxResponse{}
+	sendTxResponse.TransactionXDR = transactionXdr
+	if err != nil {
+		sendTxResponse.Code.OtherCodes = tss.RPCFailCode
+		return sendTxResponse, fmt.Errorf(err.Error())
+	}
+
+	if result, ok := rpcResponse["result"].(map[string]interface{}); ok {
+		if val, exists := result["status"].(string); exists {
+			sendTxResponse.Status = tss.RPCTXStatus(val)
+		}
+		if val, exists := result["errorResultXdr"].(string); exists {
+			sendTxResponse.Code, err = t.parseErrorResultXDR(val)
+		}
+		if hash, exists := result["hash"].(string); exists {
+			sendTxResponse.TransactionHash = hash
+		}
+	} else {
+		sendTxResponse.Code.OtherCodes = tss.RPCFailCode
+		return sendTxResponse, fmt.Errorf("RPC response has no result field")
+	}
+	return sendTxResponse, err
+}
+
+func (t *transactionService) GetTransaction(transactionHash string) (tss.RPCGetIngestTxResponse, error) {
+	rpcResponse, err := t.sendRPCRequest("getTransaction", map[string]string{"hash": transactionHash})
+	if err != nil {
+		return tss.RPCGetIngestTxResponse{Status: tss.ErrorStatus}, fmt.Errorf(err.Error())
```

Comment:
No need to wrap the error with fmt.Errorf in this case, right? But would be good to wrap it with `fmt.Errorf("sending rpc request: %w", err)` perhaps.
```suggestion
		return tss.RPCGetIngestTxResponse{Status: tss.ErrorStatus}, err
```

Reply:
Done